src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,55 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static nature() {
+    this.alive = this.alive.filter((dead) =>
+      dead.health !== 0);
+  }
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+  }
 }
+Animal.alive = [];
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+
+    Animal.alive.push(this);
+  }
+  hide(lurk = this) {
+    if (lurk.hidden !== undefined) {
+      lurk.hidden = lurk.hidden === false;
+    } else {
+      lurk.hidden = true;
+    }
```

Comment:
https://github.com/mate-academy/js_herbivores_and_carnivores/blob/master/checklist.md

Reply:
and you don't need lurk, just work with `this`